custom_components/powercalc/discovery.py

```diff
@@ -158,6 +158,10 @@ def should_process_entity(self, entity_entry: er.RegistryEntry) -> bool:
         if entity_entry.domain not in DOMAIN_DEVICE_TYPE:
             return False
 
+        if DOMAIN_DEVICE_TYPE[entity_entry.domain] == DeviceType.PRINTER:
+            if entity_entry.unit_of_measurement and entity_entry.unit_of_measurement == "%":
+                return False
```

Comment:
Not sure if this condition applicable to all printers. I will have a look at my brother printer tomorrow.
Some printers might also provide other sensors than percentage sensors.

Reply:
I thought about excluding `marker_type: ink`, maybe that's more reliable. Or only including `device_class: enum`

Reply:
Did a quick check, this is for my printer.
So status does not have a device_class in brother integration.
And there are plenty more sensors having both `%` of `p` unit of measurement.
So excluding `%` and `p` would work for both. But would it work for tens of other integrations? I don't have a clue.

But I think it's the best we can do for now.

Or we need to implement some name matching pattern on `status`, but that would not work when someone decided to rename their entity_id, and this also feels error prone.

<img width="1183" alt="Screenshot 2024-09-25 at 21 37 27" src="https://github.com/user-attachments/assets/73a2d967-e32e-4d8f-abe4-3923304872c7">

Reply:
Could just exclude if it has a unit_of_measurement attribute present. Any status sensor shouldn't have a unit of measurement.